core/imgread/common.cpp

```diff
@@ -138,158 +152,122 @@ Disc* OpenDisc(const wchar* fn)
 	return rv;
 }
 
-bool InitDrive_(wchar* fn)
+bool InitDrive_(const wchar* const fn)
 {
-	TermDrive();
+    TermDrive();
 
-	//try all drivers
-	disc = OpenDisc(fn);
+    //try all drivers
+    disc = OpenDisc(fn);
 
-	if (disc!=0)
-	{
-		printf("gdrom: Opened image \"%s\"\n",fn);
-		NullDriveDiscType=Busy;
+    if (disc)
+    {
+        printf("gdrom: Opened image \"%s\"\n", fn);
+        NullDriveDiscType = Busy;
 #ifndef NOT_REICAST
-		libCore_gdrom_disc_change();
+        libCore_gdrom_disc_change();
 #endif
-//		Sleep(400); //busy for a bit // what, really ?
-		return true;
-	}
-	else
-	{
-		printf("gdrom: Failed to open image \"%s\"\n",fn);
-		NullDriveDiscType=NoDisk; //no disc :)
-	}
-	return false;
+        return true;
+    }
+    else
+    {
+        printf("gdrom: Failed to open image \"%s\"\n", fn);
+        NullDriveDiscType = NoDisk; //no disc :)
+        return false;
+    }
 }
 
 #ifndef NOT_REICAST
-bool InitDrive(u32 fileflags)
+bool InitDrive(u32 fileflags/*=0*/)
 {
 	if (settings.imgread.LoadDefaultImage)
 	{
-		printf("Loading default image \"%s\"\n",settings.imgread.DefaultImage);
-		if (!InitDrive_(settings.imgread.DefaultImage))
+		printf("Loading default image \"%s\"\n", settings.imgread.DefaultImage.c_str());
+		if (!InitDrive_(settings.imgread.DefaultImage.c_str()))
 		{
-			msgboxf("Default image \"%s\" failed to load",MBX_ICONERROR);
+			msgboxf("Default image \"%s\" failed to load", MBX_ICONERROR);
 			return false;
 		}
 		else
 			return true;
 	}
 
-	// FIXME: Data loss if buffer is too small
-	wchar fn[512];
-	strncpy(fn,settings.imgread.LastImage, sizeof(fn));
-	fn[sizeof(fn) - 1] = '\0';
-
+	std::string diskImageFileName{ settings.imgread.LastImage };
 #ifdef BUILD_DREAMCAST
-	int gfrv=GetFile(fn,0,fileflags);
+	int gfrv = GetFile(diskImageFileName);
 #else
-	int gfrv=0;
+	int gfrv = rv_error;
```

Comment:
Should probably be initialized to `rv_ok` here?

Reply:
I'm not sure. Why we should suppose everything is OK if we forgot to assign this variable a value, for example?

Reply:
Also, tbh, I'm against the return codes. Makes code messy. We should use exceptions instead. I know we shouldn't use exceptions for controlling the code flow, but it seems to me nobody cares about return codes anyway, exceptions is harder to ignore.

Reply:
Like, all we do with return code is just propagating it up and up the call stack until we can call "exit". Not much of a logic.

Reply:
Harder to ignore, but also tend to cause more misdirected anger.